src/api/translation/open_source/open_source_translate_api.py

```diff
@@ -1,22 +1,47 @@
 from typing import List, Dict
 import pandas as pd
+from langchain_core.documents import Document
+from langchain_text_splitters import RecursiveCharacterTextSplitter
 
 from src.api.translation.i_translate_api import ITranslateAPI
 from src.api.translation.translation_result import TranslationResult
 
 
+def _combine_translation_chunks(translations: List[str], lengths: List[int]):
+    combined_translations = []
+    running_index = 0
+    for length in lengths:
+        translation_chunks = translations[running_index: running_index+length]
+        combined_translations.append(''.join(translation_chunks))
+        running_index += length
+    return combined_translations
+
+
 class OpenSourceTranslateAPI(ITranslateAPI):
-    def __init__(self, from_language: str, to_languages: List[str]):
+    def __init__(self, from_language: str, to_languages: List[str], chunk_size: int):
         super().__init__(from_language, to_languages)
+        self.text_splitter = RecursiveCharacterTextSplitter(
+            chunk_size=chunk_size,
+            chunk_overlap=20,
```

Comment:
This is problematic as it will result in overlapping text being duplicated in the final result.

Reply:
Good catch.